calico_node/startup.go

```diff
@@ -51,8 +59,239 @@ func main() {
 	// Make sure we have a global cluster ID set.
 	log.Info("Ensuring a cluster guid is set")
 	ensureClusterGuid(c)
+
+	hostname := os.GetEnv("NODENAME")
+	if hostname == nil {
+		hostname = os.GetEnv("HOSTNAME")
+	}
+	if hostname == nil {
+		hostname = os.Hostname()
+	}
+
+	if d := os.Getenv("DATASTORE_TYPE"); d == "kubernetes" {
+		f, err := os.OpenFile("startup.env", os.O_APPEND|os.O_WRONLY, 0600)
+		if err != nil {
+			panic(fmt.Sprintf("Error loading config: %s", err))
+		}
+		f.WriteString("export DATASTORE_TYPE=kubernetes\n")
+		f.WriteString("export HOSTNAME=%s\n" % hostname)
+	}
+
+	meta := api.NodeMetaData{Name: hostname}
+
+	fmt.Printf("Waiting for etcd connection...")
+	for v := os.Getenv("WAIT_FOR_DATASTORE"); v != "false"; {
+		_, err := c.Nodes().Get(meta)
```

Comment:
As far as I can tell, `c` is an instance of backend.Client, https://godoc.org/github.com/projectcalico/libcalico-go/lib/backend/api#Client, which means it doesn't actually have a Nodes() method.  We need to instantiate one of these instead: https://godoc.org/github.com/projectcalico/libcalico-go/lib/client#Client